minifi-commons/minifi-commons-schema/src/main/java/org/apache/nifi/minifi/commons/schema/serialization/SchemaLoader.java

```diff
@@ -30,13 +32,13 @@
 import java.util.stream.Collectors;
 
 public class SchemaLoader {
-    private static final Map<String, Function<Map, ConfigSchema>> configSchemaFactories = initConfigSchemaFactories();
+    private static final Map<String, Function<Map, ConvertableSchema<ConfigSchema>>> configSchemaFactories = initConfigSchemaFactories();
 
-    private static Map<String, Function<Map, ConfigSchema>> initConfigSchemaFactories() {
-        Map<String, Function<Map, ConfigSchema>> result = new HashMap<>();
-        result.put(String.valueOf((Object)null), ConfigSchema::new);
-        result.put("", ConfigSchema::new);
-        result.put("1", ConfigSchema::new);
```

Comment:
The changes in the Config schemas need to be documented. Could you add a section to the admin guide (the Config Schema portion) that explains the versioning scheme as well as what changed in v2? 

I am suggesting to put it in the docs instead of a migration guidance for a release because working with multiple different version of MiNiFi and configs will be an everyday task (esp when C&C is added).